README.md

```diff
@@ -28,6 +28,36 @@ To import it with maven, use this:
 
 ### Examples
 
+#### Cleaner transforms for Java 8.
+Java 8 introduced lambdas which can greatly reduce verbosity in code, which is great when using futures and transforms.
+
+Ideally you would want to do something like this:
+```java
+public static <A, B> ListenableFuture<B> example(ListenableFuture<A> future) {
+  return Futures.transform(future, a -> toB(a));
+}
+```
+
+This doesn't actually work though, because Futures.transform has two variants: one that takes a Function and one that takes an AsyncFunction.
+This confuses the compiler so it can't determine which variant you want to use.
```

Comment:
Maybe `Hence the compiler can't determine which variant to use without additional type information`?